workbench-core/accounts/src/services/projectService.test.ts

```diff
@@ -415,189 +500,808 @@ describe('ProjectService', () => {
           },
           ExpressionAttributeValues: {
             ':resourceType': {
               S: 'project'
-            },
-            ':proj1': { S: 'proj-123' },
-            ':proj2': { S: 'proj-123' },
-            ':proj3': { S: 'proj-123' },
-            ':proj4': { S: 'proj-123' }
+            }
           },
-          FilterExpression: 'id = :proj1 OR id = :proj2 OR id = :proj3 OR id = :proj4',
-          Limit: pageSize
+          Limit: 50,
+          ScanIndexForward: true
         })
-        .resolvesOnce(queryItemResponse1);
+        .resolves(queryItemResponse);
+
+      // OPERATE
+      const actualResponse = await projService.listProjects({
+        user,
+        sort: { status: 'asc' }
+      });
+
+      // CHECK
+      expect(actualResponse.data).toEqual([proj]);
+    });
+
+    test('list all projects as IT Admin on 1 page with sort on name', async () => {
+      // BUILD
+      const items = [projItem];
+      const queryItemResponse: QueryCommandOutput = {
+        Items: items.map((item) => {
+          return marshall(item);
+        }),
+        $metadata: {}
+      };
+      const user: AuthenticatedUser = {
+        id: 'user-123',
+        roles: ['ITAdmin']
+      };
+
+      // mock getUserGroups--TODO update after dynamic AuthZ intergration
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      jest.spyOn(ProjectService.prototype as any, '_mockGetUserGroups').mockImplementation(() => ['ITAdmin']);
+
       ddbMock
         .on(QueryCommand, {
           TableName: 'exampleDDBTable',
-          IndexName: 'getResourceByCreatedAt',
+          IndexName: 'getResourceByName',
           KeyConditionExpression: '#resourceType = :resourceType',
-          ExpressionAttributeNames: { '#resourceType': 'resourceType' },
+          ExpressionAttributeNames: {
+            '#resourceType': 'resourceType'
+          },
           ExpressionAttributeValues: {
-            ':resourceType': { S: 'project' },
-            ':proj1': { S: 'proj-123' },
-            ':proj2': { S: 'proj-123' },
-            ':proj3': { S: 'proj-123' },
-            ':proj4': { S: 'proj-123' }
+            ':resourceType': {
+              S: 'project'
+            }
           },
-          FilterExpression: 'id = :proj1 OR id = :proj2 OR id = :proj3 OR id = :proj4',
-          Limit: pageSize,
-          ExclusiveStartKey: marshall(lastEvaluatedKey)
+          Limit: 50,
+          ScanIndexForward: true
         })
-        .resolvesOnce(queryItemResponse2);
+        .resolves(queryItemResponse);
 
       // OPERATE
-      const actualResponse = await projService.listProjects({ user, pageSize });
+      const actualResponse = await projService.listProjects({
+        user,
+        sort: { name: 'asc' }
+      });
+
+      // CHECK
+      expect(actualResponse.data).toEqual([proj]);
+    });
+
+    test('list all projects as IT Admin on more than 1 page -- getting first page', async () => {
```

Comment:
There is a lot of copy and past here. How could we DRY up these tests?

Reply:
See above comment about refactoring unit tests with your help for the entire branch after feature work is merged in!